implementations/rust/ockam/ockam_identity/src/secure_channel/listener.rs

```diff
@@ -94,14 +94,29 @@ impl Worker for IdentityChannelListener {
             .options
             .create_access_control(flow_control_id.clone())?;
 
+        let credentials = if self.options.credentials.is_empty() {
+            if let Some(trust_context) = &self.options.trust_context {
+                vec![
```

Comment:
currently we are not doing something like this on the client side, I wonder if we should